pkg/fuzzer/fuzzer.go

```diff
@@ -186,29 +181,35 @@ func (fuzzer *Fuzzer) NextInput() *Request {
 			panic("queuedCandidates is out of sync")
 		}
 	}
-	if fuzzer.NeedCandidatesNow() &&
-		!fuzzer.candidatesRequested.CompareAndSwap(false, true) {
-		select {
-		case fuzzer.NeedCandidates <- struct{}{}:
-		default:
-		}
-	}
 	return req
 }
 
 func (fuzzer *Fuzzer) nextInput() *Request {
-	nextExec := fuzzer.nextExec.tryPop()
-	if nextExec != nil {
-		return nextExec.value
+	// The fuzzer may get biased to one specific part of the kernel.
+	// Periodically generate random programs to ensure that the coverage
+	// is more uniform.
+	outOfQueue := fuzzer.outOfQueue.Load() && fuzzer.outOfQueueNext.Add(1)%400 == 0
+	if !outOfQueue {
+		nextExec := fuzzer.nextExec.tryPop()
```

Comment:
It looks like we don't need priorityQueue.pop and the condvar now.

Reply:
I've sent https://github.com/google/syzkaller/pull/4597